### PR TITLE
*: Support binary operator

### PR DESCRIPTION
**Pull request by shenli**
See: https://dev.mysql.com/doc/refman/5.7/en/cast-functions.html#operator_binary

### CODE REVIEW

File: expression/expressions/cast.go [hunk 1]
```diff
@@ -22,15 +22,27 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
+// castOperatopr is the operator type for cast function.
+type castOperator int
+
+const (
+	// CastFunctionCast is CAST function.
+	CastFunctionCast castOperator = iota
```
Comment (qiuyesuifeng) [inline expression/expressions/cast.go]: It is better to use from iota + 1.

Comment (siddontang) [inline expression/expressions/cast.go]: CastFunctionCast seems confused.

Comment (qiuyesuifeng) [inline expression/expressions/cast.go]: +1

**Comment by shenli** [comment]
PTAL @siddontang @qiuyesuifeng

**Comment by siddontang** [comment]
LGTM

**Pull request merged**